keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/DeliveryConfigController.kt

```diff
@@ -125,7 +131,9 @@ class DeliveryConfigController(
       importer.import(repoType, projectKey, repoSlug, manifestPath, ref ?: "refs/heads/master")
 
     authorizationSupport.checkApplicationPermission(WRITE, APPLICATION, deliveryConfig.application)
-    authorizationSupport.checkServiceAccountAccess(SERVICE_ACCOUNT, deliveryConfig.serviceAccount)
+    deliveryConfig.serviceAccount?.also {
+      authorizationSupport.checkServiceAccountAccess(SERVICE_ACCOUNT, it)
+    }
```

Comment:
I just realized, since we're delegating to `upsert` just below, these authorization checks are unnecessary and redundant. Please remove both.

Reply:
I'm not sure that's the case for 2 reasons.

1. Internal calls within spring components do _not_ go through proxies created from annotations (if I understand correctly you're suggesting that the tests imposed by the annotations on `upsert` would be applied to the call -- they won't).
2. Delivery configs submitted directly to `upsert` might not have a `serviceAccount` so the parameter does need to be nullable in order to handle that.

Reply:
Ah, good catch. Makes sense, please disregard. Marking this as resolved.